internal/rbd/nodeserver.go

```diff
@@ -387,14 +383,14 @@ func (ns *NodeServer) stageTransaction(
 
 	err = flattenImageBeforeMapping(ctx, volOptions, cr)
 	if err != nil {
-		return transaction, err
+		return &transaction, err
 	}
 
 	// Mapping RBD image
 	var devicePath string
 	devicePath, err = attachRBDImage(ctx, volOptions, devicePath, cr)
 	if err != nil {
-		return transaction, err
+		return &transaction, err
```

Comment:
declare it as `transaction := &stageTransaction{}` so that no changes are required in the return statement?